sceptre/resolvers/__init__.py

```diff
@@ -49,62 +49,231 @@ def resolve(self):
         """
         pass  # pragma: no cover
 
+    def clone(self, stack: 'stack.Stack') -> 'Resolver':
+        """
+        Produces a "fresh" copy of the Resolver, with the specified stack.
+
+        :param stack: The stack to set on the cloned resolver
+        """
+        return type(self)(self.argument, stack)
```

Comment:
Cloning resolvers _before_ they're assigned to a stack is critical to preventing a single resolver instance being shared across multiple stacks (inherited from a stack group). The reason why that is so bad is because if the single instance is shared across Stack A and Stack B, only one of those stacks will have the right stack assigned to the resolver.